### ISSUE TITLE
Channel crashes when implementing mid roll ad-breaks  

### ISSUE THREAD

**Issue by wajidbux** [rrirower/DP2SceneGraphWizard#115]
All features in the latest update especially the intra-channel search work flawlessly but the issue of running midroll ads remains. The channel crashes when running the feed with ad breaks. The bright script errors captured on the console are given below-

![Screenshot 2024-10-06 123224](https://github.com/user-attachments/assets/fc203c4f-4a70-4135-b976-01943dd57943)

![Screenshot 2024-10-06 123301](https://github.com/user-attachments/assets/8f3b3eec-a131-4580-b2bd-cfffb247760b)

**Comment by rrirower** [comment]
Please validate your feed file with the channel feed manager.  If it finds no errors, please attach the feed to this issue. 

Note:  testing was successful with ads.  Perhaps you've found an unknown condition.

**Comment by wajidbux** [comment]
the feed file has been generated by CFM, the link is below

https://api.npoint.io/fa1104f8225ab8877442

**Comment by rrirower** [comment]
I use the Roku test ad server when testing.  I tested your channel with it and it displayed midroll ads successfully.  I'll need the URL of your ad server to test with it to see if these is an unknow condition causing the error.

**Comment by wajidbux** [comment]
vast tag-
https://tag.targeting.unrulymedia.com/rmp/215213/0/vast3?devId=ROKU_ADS_TRACKING_ID&w=ROKU_ADS_DISPLAY_WIDTH&h=ROKU_ADS_DISPLAY_HEIGHT&appbundle=ROKU_ADS_APP_ID&appname=Movieland&appstoreurl=https://channelstore.roku.com/details/2baedbba13b6d7a51e607e66298bd8a8/movieland&cntlen=ROKU_ADS_CONTENT_LENGTH&dnt=ROKU_ADS_LIMIT_TRACKING

**Comment by wajidbux** [comment]
Yes, ive reported the same problem last month as well

Yahoo Mail: Search, organise, conquer 
 
  On Mon, 7 Oct 2024 at 22:21, ***@***.***> wrote:   


This does not appear to be a new problem for you: "RAF integration throws and error" #45

**Comment by rrirower** [comment]
As I stated in the original issue:

https://github.com/rrirower/DP2SceneGraphWizard/issues/45#issuecomment-1765278517

Your ad server url does not return any ad pods that's why the code crashes.  Your initial response was that you were going to check with the ad server company.  Did you get any feedback from them?

I've tested your ad server URL and others with the [Roku Stream Tester](https://devtools.web.roku.com/stream_tester/html/) tool and it fails.  The other ad server urls work.

**Comment by wajidbux** [comment]
Yes, I contacted the ad server company; their engineers checked the tags and said it works fine. It's very strange because the same tags I used with Direct Publisher and it worked perfectly fine.

**Comment by rrirower** [comment]
The wizard code stitches the ads together on the "client side", meaning on the Roku.  Direct publisher probably cast those ads from a server.  You should ask your ad server if the URL they provided you can be used on the "client side" and stitched into the playback.  My guess is that those ads are set up to run from a server.

**Comment by wajidbux** [comment]
I emailed the ad server company again and the response is below

![Screenshot 2024-10-15 221204](https://github.com/user-attachments/assets/0db3fee1-1e93-4aa4-9d1e-376a3e424e28)
 When will this problem be solved I fail to understand

**Comment by rrirower** [comment]
Have you verified that the URL they provided contains ad pods? When I enter the URL in a browser, I get 

![Screenshot 2024-10-15 144929](https://github.com/user-attachments/assets/dd0eb5ec-ec9d-448b-86b9-1dc7c235946c)

I've posted on a developer forum to see if someone can shed some light on this.

**Comment by rrirower** [comment]
Question:  Do you always see ads when you test your channel?

**Comment by rrirower** [comment]
Everyone I've talked to says that ads should be returned.  They suggested that you explain the problem to your ad server so that they know ads are not being returned to the channel.

**Comment by wajidbux** [comment]
I have never seen any ads in the last four years. Maybe my ISP installed an ad blocker, or the ads are geo-restricted. I get my feedback from viewers and my ad server company dashboard.

**Comment by rrirower** [comment]
I got a sample ad URL from some developers.  It works with the code.  Here's what works:

`https://pubads.g.doubleclick.net/gampad/ads?iu=/21775744923/external/single_ad_samples&sz=640x480&cust_params=sample_ct%3Dlinear&ciu_szs=300x250%2C728x90&gdfp_req=1&output=vast&unviewed_position_start=1&env=vp&impl=s&correlator=`

You should have a serious talk with that ad server company, or, get a new company if you want to run ads.

**Comment by wajidbux** [comment]
> I got a sample ad URL from some developers. It works with the code. Here's what works:
> 
> `https://pubads.g.doubleclick.net/gampad/ads?iu=/21775744923/external/single_ad_samples&sz=640x480&cust_params=sample_ct%3Dlinear&ciu_szs=300x250%2C728x90&gdfp_req=1&output=vast&unviewed_position_start=1&env=vp&impl=s&correlator=`
> 
> You should have a serious talk with that ad server company, or, get a new company if you want to run ads.

there is something called a RAFX SSAI adapted to convert SSAI to CSAI stitching Google is one of them in the list I will send u the link to it
https://developer.roku.com/docs/developer-program/advertising/ssai-adapters.md

**Comment by rrirower** [comment]
If I were you, I would simply tell them that their URL is not returning any ad pods for use in client side ad stitching.

**Comment by wajidbux** [comment]
> If I were you, I would simply tell them that their URL is not returning any ad pods for use in client side ad stitching.

I guess I will do that or tie up with some other ad company

**Comment by rrirower** [comment]
Closed.  No user feedback.
